docs/designs/xref.md

```diff
@@ -91,8 +91,35 @@ xref:
   - http://url1/.xrefmap.json
   - http://url2/.xrefmap.json
 ```
+User should be able to set a string instead of array, docfx will expand it to an array during the build.\
 During `docfx restore`, all the JSON files will be restored and merged.
 
+#### Xref base path
+Since we do not have a clear picture how we are going to support the local build, we will focus on OPS build firstly.\
+The user needs to set where to get the `.xrefmap.json` files in `.openpublishing.publish.config.json`.
+```json
+{
+    "docsets_to_publish": [
+        {
+            "xref_query_tags": ["/uwp/api"]
+        }]
+}
+```
+
+#### OPS build
+With this setting in place, during the OPS build we will loop through `xref_query_tags` and get the urls of all `.xrefmap.json` files under each base path via calling DHS APIs. The reason to do so is, the public URL for each base path only contains 1 file of 1 depot with the highest priority, and there are many depots under 1 base path. These urls should only be valid for a short time.\
+\
+After getting all the urls, we will extend the DocFX config with them into `xref` settings, the one in the beginning of this section. Then, DocFX may restore the xref files as usual.
+> - The base path of current building repository should be added as default.
+> - The urls should exclude the current depot of the current buiding repository. Bucause this will be handled as internal UIDs.
```

Comment:
will this be done in build service or docfx internal?

Reply:
This logic is handled in OPS build